### PR TITLE
fix numNodesToFind

### PR DESCRIPTION
**Pull request by xiaoanyunfei**
**What type of PR is this?**
>
> /kind bug

**What this PR does / why we need it**:

if we want N Nodes,  the `checkNode` will not stop until it calculates the N + 1 result, and  then discards the (N+1)th result

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
None
```

### CODE REVIEW

**Comment by k82cn** [comment]
please add related test for it.

File: pkg/scheduler/core/generic_scheduler.go [hunk 1]
```diff
@@ -473,12 +473,10 @@ func (g *genericScheduler) findNodesThatFit(pod *v1.Pod, nodes []*v1.Node) ([]*v
 				return
 			}
 			if fits {
+				filtered[length] = g.nodeInfoSnapshot.NodeInfoMap[nodeName].Node()
```
Comment (denkensk) [inline pkg/scheduler/core/generic_scheduler.go]: length is used before declaration.  If I understand correctly, you want to use filteredLen?

**Comment by xiaoanyunfei** [comment]
> please add related test for it.
`checkNode` is an inside function, is there any good suggestions for testing it?

**Comment by k82cn** [comment]
> checkNode is an inside function, is there any good suggestions for testing it?

Add an unit test for it, at least :)

**Comment by xiaoanyunfei** [comment]
en, after using an unit test, I found `checkNode ` could not stop in time when using multi goroutine. It calculates M (M<10) more nodes than excepted N

**Pull request closed without merging**